src/lib/rest/rest.cpp

```diff
@@ -616,6 +630,17 @@ static int contentTypeCheck(ConnectionInfo* ciP)
     return 1;
   }
 
+
+  // Case 5
+  if ((ciP->apiVersion == "v2") && (ciP->httpHeaders.contentType != "application/json"))
```

Comment:
> XML is still the default format.
> If we change that ... things may stop working for some people ...

What about make the default depend on the API version. I mean, in the case of /v2 the default is JSON. In the case of not-v2, the default is XML to mantain backward compatibility.